cmd/controller-manager/app/controllermanager.go

```diff
@@ -129,6 +132,7 @@ func Run(ctx context.Context, opts *options.Options) error {
 		LivenessEndpointName:       "/healthz",
 		MetricsBindAddress:         opts.MetricsBindAddress,
 		MapperProvider:             restmapper.MapperProvider,
+		BaseContext:                reUseContext,
```

Comment:
Wouldn't it be better to write it this way?
```go
BaseContext: func() context.Context { return ctx }
```

Reply:
yep , that's great.
thanks.